### DISCUSSION TITLE
[Feature] Server url QR code for app

### DISCUSSION THREAD

**Discussion by rdaniel0** [immich-app/immich#14763] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

## I would like to see
Improve UX around server selection by providing more friendly completion of `server-url` entry

### Why?
This is particularly aimed at non-tech literate users

This would reduce confusion by presenting the user with something familiar (a qr code they can scan on their camera app) that would copy or deep-link into the app to alleviate `server-url` entry and potentially take them straight to the login screen (a more familiar concept for most users)

## Proposed criteria
1. Web/Server UI: An option somewhere to show server's url (using whatever the is in the browser's address bar?) as a QR code
  1a. Bonus: Encode url as a deeplink to the app server url entry screen
2. App: On server url screen, add an option to scan a QR code.
  2a. Bonus:  Deep-link detection opens the app and fills in server url and takes user to login screen. Would need to handle logout and other edge cases.

## Notes
I found this similar `issue` https://github.com/immich-app/immich/issues/655 also asking for qr codes for both server-url + login, but this request is specifically for just the server url
This would simplify any security concerns since we're not trying to encode authentication

Potentially duplicate `discussion` https://github.com/immich-app/immich/discussions/6998 but this doesn't contain enough info to get started

If you agree to any of the above points, I'm happy to learn and make a PR given some direction.
I can already see there is an open (approved, not yet merged?) PR for qr codes on the web app https://github.com/immich-app/immich/pull/13128 so we can reuse that work (assuming this gets merged at some point).

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Thanks for the FR! I think this is a duplicate of #6998, even if that doesn't have as much info as you've posted here - that info can be added on that request as well ;)

There's also #8766, which I think would be a basis for this feature to build on top of. 

A PR would be welcome, thank you!
